_minutes/2023-07-20-wecg.md

```diff
@@ -0,0 +1,165 @@
+
+ * [anton] Different browsers have different optional permission support. Can all permissions be optional? E.g. the “devtools” permission is effectively a remote code permission, which should ideally be optional.
+ * [rob] I have some additional context. But first, is devtools an example or the primary permission you're interested in?
+ * [anton] Just an example. contextMenus as an example, users cannot disable them.
+ * [rob] Context menu case is different as it doesn't have a permission warning. Might be interesting to explore what we want to do with permissions that do not have a permission warning. For “devtools”, in Firefox this makes the devtools_page optional in firefox. In Chrome, it displays a scary warning. Whether APIs are optional or non-optional is primarily an API design detail and partly implementation. Revoking/granting API permissions can be expensive in some cases (e.g. DNR requires expensive computation).
+ * [simeon] I want to encourage the platform to move in the direction of optional permissions.
+ * [tomislav] Most of our APIs have optional permissions, with exceptions that Rob mentioned. To the topic of the “contextMenus” permission, which doesn't have a permission warning: revoking the permission and disabling an API may result in unexpected behavior to extensions. If others want opt-outs for promptless permissions, we'd be open to consider it.
+ * [timothy] We auto-grant API permissions without permission prompts. Everything is gated on host permissions.
+ * [sam] To echo from the DuckDuckGo side: existing extensions with large user bases cannot use permissions with warnings, because that prevents users from upgrading smoothly. Optional permissions or prompt-less grants help with this use case.
+ * [simeon] Expanding on that, in Chrome new required permissions will cause the extension to be disabled on update until the user approves any newly requested permissions.
+ * [timothy] API permission requests are granted silently, but there are some APIs where a new request results in disabling the extension, if they don't overlap with the bucket of similar capabilities.
+ * [rob] If permissions.request is used with a different-bucket extension, does that also disable the extension? Or is it silently granted?
+ * [timothy] Not sure, I have to check.
+ * [tomislav] In Firefox we don't disable the extension, but we don't upgrade it. The user is prompted to confirm new permissions and upgrade is delayed until they approve the request.
+ * [simeon] Pausing discussion in the interest of briefly discussing the final issue.
+
+[Issue 426](https://github.com/w3c/webextensions/issues/426): Inconsistency: Navigation to file:// URLs using tabs and windows APIs
+
+ * [oliver] Security concerns with navigating to file:-URLs from extensions. Concern is less about abuse by extensions, more about extensions tricking other extensions into navigating to file:-URLs.
```

Comment:
```suggestion
 * [oliver] Security concerns with navigating to file:-URLs from extensions. Concern is less about abuse by extensions, more about vulnerable extensions being tricked into navigating to file:-URLs.
```

Reply:
@Rob--W, can we merge this here or would you prefer to merge locally and force push?